src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1710,14 +1697,12 @@ void enableSSL(String host,
                 tmf.init(ks);
                 tm = tmf.getTrustManagers();
 
-                // if the host name in cert provided use it or use the host name Only if it is not FIPS
-                if (!isFips) {
-                    if (null != hostNameInCertificate) {
-                        tm = new TrustManager[] {new HostNameOverrideX509TrustManager(this, (X509TrustManager) tm[0], hostNameInCertificate)};
-                    }
-                    else {
-                        tm = new TrustManager[] {new HostNameOverrideX509TrustManager(this, (X509TrustManager) tm[0], host)};
-                    }
+                // if the host name in cert provided use it or use the host name 
```

Comment:
Thank you for pointing this out. I am able to reproduce the exception, will investigate and get back to you.